ui/cap-react/src/components/drafts/DraftsRouteGuard.js

```diff
@@ -18,11 +20,19 @@ const DraftsRouteGuard = props => {
       false: (
         <RouteGuard
           when={true}
-          navigate={path => props.history.push(path)}
+          navigate={path => {
+            // update location
+            props.history.push(path);
+
+            if (!location.pathname.startsWith(`/drafts/${props.draft_id}`)) {
+              props.initDraftState();
```

Comment:
this will only re-init draft state when condition is false...we need this every time